src/wallet/bdk.rs

```diff
@@ -103,11 +103,40 @@ impl KeychainKind {
         match self {
             Self::External => 0,
             Self::Internal => 1,
-            Self::Fidelity => 2,
-            Self::SwapCoin => 3,
-            Self::Contract => 4,
+            Self::Fidelity { .. } => 2,
+            Self::SwapCoin { .. } => 3,
+            Self::Contract { .. } => 4,
         }
     }
+
+    /// return Fidelity index no
+    pub fn fidelity_kind_index() -> u32 {
+        KeychainKind::Fidelity { count: 0 }.index_num()
+    }
+
+    /// returns Contract index no
+    pub fn contract_kind_index() -> u32 {
+        KeychainKind::Contract { count: 0 }.index_num()
+    }
+
+    /// returns Swapcoin index no
+    pub fn swapcoin_kind_index() -> u32 {
+        KeychainKind::SwapCoin { count: 0 }.index_num()
+    }
+
+    /// create Fidelity keychain with given count no
+    pub fn create_fidelity_keychain(count: u32) -> Self {
+        KeychainKind::Fidelity { count }
+    }
+
+    /// create Contract keychain with given count no
+    pub fn create_contract_keychain(count: u32) -> Self {
+        KeychainKind::Contract { count }
+    }
+    /// create Swapcoin keychain with given count no
+    pub fn create_swapcoin_keychain(count: u32) -> Self {
+        KeychainKind::SwapCoin { count }
+    }
```

Comment:
We don't need this functions. We can as well create key chain by `let x = KeychainKind::Contract { 42 }`.